src/System.Runtime/tests/System/GCTests.netcoreapp.cs

```diff
@@ -35,5 +36,59 @@ public static void GetGCMemoryInfo()
             Assert.True(memoryInfo.HeapSizeBytes > 0);
             Assert.True(memoryInfo.FragmentedBytes >= 0);
         }
+
+        [Fact]
+        public static void GetGCMemoryInfoFragmentation()
+        {
+            GCHandle[] gch = null;
+
+            try
+            {
+                gch = new GCHandle[100];
+
+                for (int i = 0; i < gch.Length * 2; ++i)
+                {
+                    byte[] arr = new byte[128];
+                    if (i % 2 == 0)
+                    {
+                        gch[i / 2] = GCHandle.Alloc(arr, GCHandleType.Pinned);
+                    }
+                }
+
+                GC.Collect();
+
+                GCMemoryInfo memoryInfo = GC.GetGCMemoryInfo();
+
+                Assert.True(memoryInfo.FragmentedBytes >= 128 * (gch.Length - 1), $"FragmentedBytes = {memoryInfo.FragmentedBytes}");
+            }
+            finally
+            {
+                for (int i = 0; i < gch.Length; ++i)
```

Comment:
Very unlikely to happen, but if the `new GCHandle[100]` OOMs, this will null ref.  The allocation of the array should just be moved up to the declaration, e.g. `var gch = new GCHandle[100];`, outside of the try.